src/main/java/com/jagrosh/discordipc/entities/Callback.java

```diff
@@ -39,46 +39,34 @@ public Callback()
     }
 
     /**
-     * Constructs a Callback with a success {@link Runnable} that
+     * Constructs a Callback with a success {@link Consumer} that
      * occurs when the process it is attached to executes without
      * error.
      *
-     * @param success The Runnable to launch after a successful process.
+     * @param success The Consumer to launch after a successful process.
      */
-    public Callback(Runnable success)
+    public Callback(Consumer<Packet> success)
     {
         this(success, null);
     }
 
     /**
-     * Constructs a Callback with a failure {@link Consumer} that
-     * occurs when the process it is attached to encounters an error,
-     * and whose parameter is the error message.
-     *
-     * @param failure The Consumer to launch if the process has an error.
-     */
-    public Callback(Consumer<String> failure)
-    {
-        this(null, failure);
-    }
-
-    /**
-     * Constructs a Callback with a success {@link Runnable} <i>and</i>
+     * Constructs a Callback with a success {@link Consumer} <i>and</i>
      * a failure {@link Consumer} that occurs when the process it is
      * attached to executes without or with error (respectively).
      *
-     * @param success The Runnable to launch after a successful process.
+     * @param success The Consumer to launch after a successful process.
      * @param failure The Consumer to launch if the process has an error.
      */
-    public Callback(Runnable success, Consumer<String> failure)
+    public Callback(Consumer<Packet> success, Consumer<String> failure)
```

Comment:
I've made a slight reconsideration, I'd like you to reimplement the constructors that accept Runnable and have them provide the runnable code to a consumer, as well as deprecate them so they can be reasonably removed in a future release with proper warning.
```java
@Deprecated
public Callback(Runnable success, Consumer<String> failure)
{
    this(p -> success.run(), failure);
}
```

While this certainly creates more overhead in the next release, I want to give library users the time they need to move to the new constructor.
Finally, make sure to provide a proper `@deprecated` tag in the javadoc comments for these constructors.

Most likely we'll remove them in the version *after* this next release, but for now I'd like to avoid any catastrophic compiler errors in the mean time.

Reply:
I attempted this but it was impossible due to type erasure in the constructor parameter

Reply:
Uh, are you referring to calling `this(null)` and `this(null, failure)`?
If you are, simply rework each constructor that had type erasure issues so that it set the fields manually, it compiles fine: 
```java
// Doesn't work
public Callback(Consumer<String> failure)
{
    this(null, failure);
}

// Works
public Callback(Consumer<String> failure)
{
    this.success = null;
    this.failure = failure;
}
```

With that, this will be modified once we remove the constructors, but for now it's possible so I'd appreciate it if changes were made (unless I'm completely misunderstanding what "type erasure" is in this context).

Reply:
No, I mean you can't have `Callback(Consumer<String>)` and `Callback(Consumer<Packet>)`, since that would mean having two `Callback(Consumer)` as the type is "erased".

Reply:
I guess the only compromise here would be to remove `Consumer<String>` and retain + deprecate as I've requested then. 🤔